### ISSUE TITLE
False positive for no-unused-vars

### ISSUE THREAD

**Issue by florianwittmann** [oxc-project/oxc#4888]
In the following case _text_ in the first line is found by no-unused-vars, but not found by the original eslint rule:
```
   const { text, ...dbEntry } = entry;
    return doSomething({
      ...dbEntry,
      someOtherProp
    });
```
Here we want to get _entry_ without _text_ and this seems to be one legit way to do this, hence _text_ is **not unused**. This was tested with version 0.7.1.

**Comment by camc314** [comment]
hmm this reports an error on `text` for me? maybe a diff between the eslint vs typescript eslint rule?

the quick solve for this is to do something like: `const { text: _text, ...dbEntry } = entry` although it requires changing code

<img width="1074" alt="Screenshot 2024-08-14 at 09 18 53" src="https://github.com/user-attachments/assets/185850e8-831a-4ebe-a282-75dfc66a13fd">

**Comment by DonIsaac** [comment]
I think there's a bug in the code handling ignoreRestSiblings. 

Just to confirm, you did set this option to `true`, correct?

**Comment by florianwittmann** [comment]
Yes, it is active, with the following additional configuration:
```
      "no-unused-vars": [
        "warn",
        {
          "args": "none",
          "caughtErrors": "none",
          "ignoreRestSiblings": true,
          "vars": "all"
        }
      ],
```

**Comment by DonIsaac** [comment]
@florianwittmann can you please show me the command you're using to run oxlint?

**Comment by DonIsaac** [comment]
I am unable to reproduce this issue, please provide any additional setup/command arguments/etc you used so I can debug it.

**Comment by florianwittmann** [comment]
Found the problem, it was referencing the wrong config file the hole time 🤦 Sorry, everything seems to work as intended, when configured correctly.
